server/src/auth-router.ts

```diff
@@ -119,34 +119,32 @@ router.get(
         session.passport.user = old_user;
       }
       response.redirect('/profile/settings?success=' + success.toString());
-    } else if (enrollment && enrollment.challenge && enrollment.team) {
-      if (
-        !(await UserClient.enrollRegisteredUser(
-          user.emails[0].value,
-          enrollment.challenge,
-          enrollment.team,
-          enrollment.invite,
-          request.header('Referer')
-        ))
-      ) {
-        // if the user is unregistered, pass enrollment to frontend
-        user.enrollment = enrollment;
-      } else {
-        // if the user is already registered, now he/she should be enrolled
-        // [TODO] there should be an elegant way to get the client_id here
-        const client_id = await UserClient.findClientId(user.emails[0].value);
-        await earnBonus('sign_up_first_three_days', [
-          enrollment.challenge,
-          client_id,
-        ]);
-        await earnBonus('invite_signup', [
-          client_id,
-          enrollment.invite,
-          enrollment.invite,
-          enrollment.challenge,
-        ]);
-      }
-
+    } else if (
+      enrollment &&
+      enrollment.challenge &&
+      enrollment.team &&
+      (await UserClient.enrollRegisteredUser(
+        user.emails[0].value,
+        enrollment.challenge,
+        enrollment.team,
+        enrollment.invite,
+        request.header('Referer')
+      ))
+    ) {
+      // if the user is already registered, now he/she should be enrolled
+      // [TODO] there should be an elegant way to get the client_id here
+      const client_id = await UserClient.findClientId(user.emails[0].value);
+      user.enrollment = enrollment; // TODO: Do we need this?
```

Comment:
Seconding Q of whether we need this @can-007

Reply:
I think we need this.
This enroll function here is for enrolling registered user only.
For brand new user, the enrollment happens when saving the profile. And the enrollment information will be needed then.

I think this PR will prevent brand new user from enrolling. He/she has to enroll again to get enrolled here.